### ISSUE TITLE
Feature Request: add rule to detect circular dependency injection

### ISSUE THREAD

**Issue by ptandler** [angular-eslint/angular-eslint#2082]
**Description and reproduction of the issue**

In a large Angular app it can happen that circular dependency injection is introduced - and this leads to run-time errors. It would be very hand to get an error already during static code analysis.

The simplest case that should be detected is this:

```ts
// a.service.ts
@Injectable()
export class A {
  private readonly b = inject(B);
}

// b.service.ts
@Injectable()
export class B {
  private readonly a = inject(A);
}

But of course, it's not as easy to track down if you have more services in between.

```

```JSON
{
  "rules": {
      "@angular-eslint/no-circular-dependency-injection": "error"
  }
}
```

**Comment by JamesHenry** [comment]
Hi @ptandler I strongly feel this should be a feature of the framework. As you have said, statically analyzing this is tricky anyway, and it definitely seems like something which should exist in Angular itself, because whether or not circular dependencies are an issue in practice is related to the semantics of `inject()`.

**Comment by ptandler** [comment]
wow, that was a quick reply! :slightly_smiling_face: 
So I'll add an FR there.
